### PR TITLE
Create docker image for deploying TCP to BLOB

### LINKED ISSUE #143: TCP to BLOB: Create Docker Image with deployment dependencies
Pulled from Hrishi's comments in "Orbital Integration - Orbital Analytics and Data" on 9/8:

"Small ask - is it possible to make a docker image with Node JS, yarn, Azure CLI preloaded? Or a script? The user will most likely get the wrong CLI by using the default 'sudo apt install' method. Same comment applies to Node JS. Furthermore, Bastion is having copy past issues with some of the cmd line commands located on the installation docs for these packages"

### PR DESCRIPTION
**Pull request by micholas**
# Description
Fixes #143

1. Create Docker file for building docker image with dependencies needed for deploying TCP to BLOB.
2. Update README to describe how to deploy TCP to BLOB using the deployer docker image.

## Dependencies affected:
- n/a

## Checklist before merging
- [x] Properly labeled PR 
- [x] Licensing statement added to new files 
- [x] Downstream dependencies have been addressed
- [x] Corresponding changes to the documentation have been made
- [x] Issue is linked under the development section

### CODE REVIEW

File: tcp-to-blob/deploy/env-template.sh [hunk 1]
```diff
@@ -5,5 +5,6 @@
 
 # The following is a template for the required env vars.
 # Please see the README for more details
+export AZ_RESOURCE_GROUP="$AZ_RESOURCE_GROUP"
```
Comment (scschneider) [inline tcp-to-blob/deploy/env-template.sh:8]:
```suggestion
```
This is not required

File: tcp-to-blob/deploy/deploy-in-docker.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+set -euo pipefail
+
+az account show # Fail fast if not logged in.
+PROJECT_DIR="$(dirname -- "$0")/../.."
+echo "PROJECT_DIR: \"${PROJECT_DIR}\""
+pushd "${PROJECT_DIR}"
+./install-node-modules.sh
+npx yarn build
+mkdir tcp-to-blob/.env
```
Comment (scschneider) [inline tcp-to-blob/deploy/deploy-in-docker.sh:14]:
```suggestion
mkdir -p tcp-to-blob/.env
```
If you re run the deploy, this errors out saying the dir already exists. Using -p will keep that error from popping up.

File: tcp-to-blob/deploy/Dockerfile_deployer [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ARG IMAGE_NAME=mcr.microsoft.com/azure-cli
+
+FROM $IMAGE_NAME
+
+WORKDIR /home/
+
+RUN apk add --update nodejs npm zsh git vim gettext
+
+RUN az aks install-cli
+RUN sh -c "$(wget https://raw.github.com/robbyrussell/oh-my-zsh/master/tools/install.sh -O -)"
```
Comment (scschneider) [inline tcp-to-blob/deploy/Dockerfile_deployer:18]:
```suggestion
RUN az extension add --name portal
RUN sh -c "$(wget https://raw.github.com/robbyrussell/oh-my-zsh/master/tools/install.sh -O -)"
```
Proactively install extension portal so users are not prompted to do so.

File: tcp-to-blob/README.md [hunk 4]
```diff
@@ -104,23 +92,60 @@ Optional:
...
+2. `docker build . -f deploy/Dockerfile_deployer -t orbital-integration-deployer`
+3. `NAME_PREFIX=<desired_name_prefix>` Set prefix for names of resources to be deployed.
+4. `docker run -it -e NAME_PREFIX orbital-integration-deployer:latest`
+5. The command above will bring you to a container shell. In container shell:
+   1. `az login`
+   2. `az account set -s <your_subscription>`
+   3. `git pull`
+   4. (optional) Update .env/env-template.sh if desired. See Environment Variables section above.
+   5. `./tcp-to-blob/deploy/deploy-in-docker.sh`
+   6. If any commands ask to install extensions, type "y".
```
Comment (scschneider) [inline tcp-to-blob/README.md:116]: Not required if we install `az extension add --name portal` during docker build.

**Comment by scschneider** [comment]
The documentation needs a little updating regarding flow of steps. We sill do `Install NodeJS dependencies & build project`, but I do not see where this step is required now that we are deploying within a Docker container.

In the docs, we create an .env file, but never use it within the Docker container. I think we can and should still leverage this.

File: tcp-to-blob/README.md [hunk 5]
```diff
@@ -104,23 +92,60 @@ Optional:
...
+
+- [Docker](https://command-not-found.com/docker)
+
+#### Procedure
 
 1. `git clone https://github.com/Azure/azure-orbital-integration.git`
-2. If using [Azure Cloud Shell](https://learn.microsoft.com/en-us/azure/cloud-shell/overview), you are already logged in. Otherwise, Ensure [logged in](https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli) to Azure CLI and default
+2. `docker build . -f deploy/Dockerfile_deployer -t orbital-integration-deployer`
+3. `NAME_PREFIX=<desired_name_prefix>` Set prefix for names of resources to be deployed.
+4. `docker run -it -e NAME_PREFIX orbital-integration-deployer:latest`
```
[the remark this answers is not in the document]
Comment (micholas) [reply tcp-to-blob/README.md:110]: https://docs.docker.com/engine/reference/run/

> If the operator names an environment variable without specifying a value, then the current value of the named variable is propagated into the container’s environment

[the remark this answers is not in the document]
Comment (scschneider) [reply tcp-to-blob/README.md:110]: I just sent a screenshot of my experience when I tried this and I still had to pass in the NAME=VALUE

**Review by scschneider — APPROVED** [review]

**Pull request merged**